A38-admin-interface-api.md

```diff
@@ -0,0 +1,200 @@
+Admin Interface API
+----
+* Author(s): lidiz
+* Approver: markroth
+* Status: Draft
+* Implemented in: C++, Golang, Java
+* Last updated: 2021-01-27
+* Discussion at: TBD
+
+## Abstract
+
+This proposal describes a convenience API in each gRPC language to improve the usability of creating a gRPC server with admin services.
+
+
+## Background
+
+gRPC Admin Interface is part of the Proxyless Service Mesh Observability project which includes new gRPC admin services and a CLI tool (yet to release). The admin services are responsible for exposing states of the gRPC library via gRPC protocol, and the CLI tool will help users to better interpret the fetched data from the gRPC application. Currently, the existing admin server should at least expose two admin services:
```

Comment:
I don't think we should mention PSM or observability here, since those are not OSS concepts.  I think this gRFC should not mention anything xDS-specific at all.  We should have a separate gRFC for CSDS (which should talk about xDS, not about PSM or observability), and the two gRFCs can refer to each other.

For this gRFC, I suggest just focusing on the following key points:
- There exists a class of RPC services whose purpose is to expose the overall state of all gRPC activity in a given binary.  This set of services may grow over time.
- Some services may want to be added conditionally, to avoid adding unnecessary dependencies (e.g., CSDS should be used only when xDS is used in gRPC).
- We want to provide a mechanism where applications can tell gRPC to add all such services to a given server, and gRPC will automatically add all appropriate services, so that individual applications don't need to change whenever the desired set of admin services changes.

Reply:
(I thought we named the xDS effort "proxyless service mesh", which is also Menghan's presentation title at KubeCon.)

Restructured the design doc.